rasa/shared/core/domain.py

```diff
@@ -225,15 +225,17 @@ def _get_session_config(session_config: Dict) -> SessionConfig:
     @classmethod
     def from_directory(cls, path: Text) -> "Domain":
         """Loads and merges multiple domain files recursively from a directory tree."""
-
-        domain = Domain.empty()
+        domain_dict = {}
         for root, _, files in os.walk(path, followlinks=True):
             for file in files:
                 full_path = os.path.join(root, file)
                 if Domain.is_domain_file(full_path):
-                    other = Domain.from_file(full_path)
-                    domain = other.merge(domain)
```

Comment:
What's the difference between `Domain.merge` vs `Domain.merge_domain_dicts`?

Reply:
Domain.merge() is a method that merges an already parsed Domain object onto another already parsed domain object.

Domain.merge_domain_dicts() is a _static_ method that merges to "raw" domain dicts _before_ they get converted into a Domain object (in which process the entity details are lost: One-out-of-one == all)

Reply:
The culprit is `Domain.from_dict()` and similar here. In there the use_entities is replaced by "used_entities" which lacks some of the original information.

Reply:
So we do that as last action, after merging the original unchanged data. See line 238 below.

Reply:
I see what you mean @raoulvm thanks. I'm trying to think of an elegant way to fix this issue. I feel there's code duplication with two `merge` methods but in the end I guess we can remove the old `merge` method and just keep the new `merge_domain_dicts`. It makes sense to create Domain after dictionaries are merged.

Reply:
>  I feel there's code duplication with two merge methods

Like "a lot" because that is the way I created it...